generator/tests/test_other_library/run_other_library_tests.py

```diff
@@ -323,44 +422,75 @@ def run_enum(name, class_name, test_case):
+    """
+    Compare the 'bindings' files.
+
+    :param name: stub of XML filename, e.g. 'test_sedml' for file test_sedml.xml.
+    :param class_name: the test class, e.g. 'libsedml'
+    :param test_case: brief description of tests, e.g. 'swig dir'
+    :param binding: the binding type, e.g. 'swig'
+    :param prefix: used to match library files, e.g. 'combine' matches libcombine-*.*
+    :returns: number of failed tests.
+    """
     filename = test_functions.set_up_test(name, class_name, test_case)
     generate_binding(filename, binding)
     fail = 0
     if binding == 'swig':
-        fail += compare_binding_headers('ListWrapper', binding, prefix)
-        fail += compare_binding_headers('OStream', binding, prefix)
-        fail += compare_binding_impl('OStream', binding, prefix)
-        fail += compare_binding_interface('std_set', binding, prefix)
-        fail += compare_binding_headers('lib{0}'.format(prefix), binding, prefix)
-        fail += compare_binding_interface('lib{0}'.format(prefix), binding, prefix)
-        fail += compare_binding_interface('ASTNodes', binding, prefix)
+        fail += compare_binding_headers('ListWrapper', binding, "")
+        fail += compare_binding_headers('OStream', binding, "")
+        fail += compare_binding_impl('OStream', binding, "")
+        fail += compare_binding_interface('std_set', binding, "")
+        fail += compare_binding_headers('lib{0}'.format(prefix), binding, "")
+        fail += compare_binding_interface('lib{0}'.format(prefix), binding, "")
+        fail += compare_binding_interface('ASTNodes', binding, "")
     elif binding == 'csharp':
-        fail += compare_binding_impl('local', binding, prefix)
-        fail += compare_binding_interface('local', binding, prefix)
-        fail += compare_binding_interface('lib{0}'.format(prefix), binding, prefix)
-        fail += compare_binding_file('CMakeLists.txt', binding, prefix)
-        fail += compare_binding_file('compile-native-files.cmake', binding, prefix)
-        fail += compare_binding_file('AssemblyInfo.cs.in', binding, prefix)
+        fail += compare_binding_impl('local', binding, "")
+        fail += compare_binding_interface('local', binding, "")
+        fail += compare_binding_interface('lib{0}'.format(prefix), binding, "")
+        fail += compare_binding_file('CMakeLists.txt', binding, "")
+        fail += compare_binding_file('compile-native-files.cmake', binding, "")
+        fail += compare_binding_file('AssemblyInfo.cs.in', binding, "")
     print('')
     return fail
 
-def test_cmake(name, class_name, test_case, binding, prefix):
+
+def test_cmake(name, class_name, test_case, binding):  # , prefix):
+    """
+    Generate and compare CMake files.
+
+    TODO the following need more details:
+
+    :param name: name of test, e.g. 'test_sedml' or 'combine-archive'.
+    :param class_name:   e.g. 'libsedml' or 'libcombine'
+    :param test_case: seems to always be 'cmake'!
```

Comment:
test_case and binding could be removed as parameters and hardcoded as 'cmake'

Reply:
OK, removed parameters